lib/phoenix_live_view/channel.ex

```diff
@@ -613,6 +615,19 @@ defmodule Phoenix.LiveView.Channel do
     {:stop, :shutdown, :no_session}
   end
 
+  defp check_version_match(params) do
+    server_version = Application.spec(:phoenix_live_view, :vsn) |> List.to_string()
+    client_version = Map.get(params, "lv_version", :not_found)
+    should_display_warning = Application.get_env(:phoenix_live_view, :warn_version_mismatch, true)
+
+    if should_display_warning && server_version != client_version do
+      Logger.warn("""
+      There is a mismatch between the LiveView version on the server (#{server_version}) and the client (#{client_version}).
+      The client side can be synced to match by running `npm install --prefix assets`.
```

Comment:
I believe we should include the name of the LiveView we are trying to execute here. That's because sometimes the client is embedded externally, such as in the LiveView Dashboard. So it would be good to know if the error is coming from an external application or our own application. WDYT?

Reply:
Yes, on this one I completely agree, the more specific this is the better, imo. I'm not 100% sure how to do this but will give it a try.